src/Components/InstanceCounter/index.js

```diff
@@ -37,18 +39,34 @@ const InstanceCounter = () => {
   };
 
   return (
-    <NumberInput
-      max={MAX_INSTANCES}
-      min={MIN_INSTANCES}
-      value={chosenNumOfInstances}
-      onMinus={onMinus}
-      onChange={onChange}
-      onPlus={onPlus}
-      inputName="instances"
-      inputAriaLabel="number of instances"
-      minusBtnAriaLabel="instances-minus"
-      plusBtnAriaLabel="instances-plus"
-    />
+    <>
+      {!(vCPUCount * chosenNumOfInstances <= 32) && (
+        <Alert
+          data-testid="vcpu_limit_exceeded_alert"
+          ouiaId="vcpu_limit_exceeded_not_supported_alert"
+          variant="warning"
+          isInline
+          title="Warning: vCPU is over the limit with this amount of instances."
+        />
+      )}
+      <NumberInput
+        max={MAX_INSTANCES}
+        min={MIN_INSTANCES}
+        value={chosenNumOfInstances}
+        onMinus={onMinus}
+        onChange={onChange}
+        onPlus={onPlus}
+        inputName="instances"
+        inputAriaLabel="number of instances"
+        minusBtnAriaLabel="instances-minus"
+        plusBtnAriaLabel="instances-plus"
+      />
+    </>
   );
 };
+
+InstanceCounter.propTypes = {
+  vCPUCount: PropTypes.number.isRequired,
```

Comment:
sounds weird to have this required as it doesn't feel too necessary for the component, could we allow to omit it with default `null` and hide the warning when not passed (when value is null)?
If you want to cheat, default `0` will always `<= 32` ofc :)